docs/firststeps.rst

```diff
@@ -1,35 +1,129 @@
+*CliMetLab* provides two main ways to access climate data
+
+1. Data Sources
+2. Datasets
 
 Data sources
 ^^^^^^^^^^^^
 
-Data sources implement various methods to access and decode data.
-When data are downloaded from a remote site, they are :ref:`cached
-<caching>` on the local computer.
+In *CliMetLab*, a *Data Source* refers to a local or remote storage server or data archive
+from where we can download or access files related to climate data.
+
+To get started, let us first import *CliMetLab* at the top of our Python notebook:
+
+.. code-block:: python
+
+    import climetlab as cml
+
+Now download
+`test.grib <https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.grib>`_ (example GRIB file)
+to your project directory, or if you have ``wget`` command available, run
+following in your notebook:
+
+.. code-block:: python
+
+    !wget https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.grib
+
+GRIB is a file format for storage and sharing of gridded meteorological data.
+You can think of gridded data as weather or some other data that is associated with
+specific locations using coordinates. For example, wind speed data for
+every longitude and latitude on a two dimensional grid.
+
+GRIB is designed and maintained by World Meteorological Organization as an
+easy to handle and compact data format. GRIB (GRIdded Binary) is a binary file
+format so you cannot look at it using a text editor. But you sure can use
+*CliMetLab* to explore it:
+
+.. code-block:: python
+
+    grib_data = cml.load_source("file", "test.grib")
+
+Here we used ``load_source`` method from *CliMetLab* to load our GRIB file
+into ``grib_data`` variable. The first argument ``"file"`` specifies the type of
+our data source. Which currently is indeed a local file that we downloaded.
+The second argument is the path to that file.
+
+Let's plot this data using the ``plot_map`` convenience method:
+
+.. code-block:: python
+
+    cml.plot_map(data, title=True, path="test-grib-plot.svg")
+
+
+.. image:: _static/example-plots/test-grib-plot.svg
+  :width: 100%
+
+The ``title`` and ``path`` arguments supplied to ``plot_map`` are optional.
+When ``title`` is set to ``True``, the plot will include the title of the data
+from our data file. If you specify the
+``path`` ending with a supported format (``.svg``, ``.png``, ``.pdf``), the
+plot will be saved at the specified location.
+
+You can also use ``load_source`` to directly download and load files from remote
+server, in one step. For example, let's download and plot a NetCDF file:
+
+.. code-block:: python
+
+    netcdf_url = "https://raw.githubusercontent.com/ecmwf/climetlab/develop/docs/examples/test.nc"
+    netcdf_data = cml.load_source("url", netcdf_url)
+    cml.plot_map(netcdf_data)
+
+.. image:: _static/example-plots/test-netcdf-plot.svg
+  :width: 100%
+
+If you don't know already, NetCDF is another commonly used data format for
```

Comment:
Add a link to netcdf, maybe?

Reply:
A [Wikipedia](https://en.wikipedia.org/wiki/NetCDF) or link to [NetCDF docs](https://www.unidata.ucar.edu/software/netcdf/docs/)?